### PR TITLE
feat(locale): add states to vi locale and fix city name

### PR DESCRIPTION
**Pull request by asharnadeem**
Addresses concerns in #1977. Adds `state` support for the `vi` locale, and as part of this PR, also fixes `city_name` where the previous list included both cities and states. This PR introduces more realistic location data overall for the `vi` locale

### CODE REVIEW

File: src/locales/vi/location/state.ts
Comment (matthewmayer) [inline src/locales/vi/location/state.ts:1]: Should these include diacritics, like the city names? For example [Bình Phước](https://vi.wikipedia.org/wiki/B%C3%ACnh_Ph%C6%B0%E1%BB%9Bc) not `Binh Phuoc`

**Review by matthewmayer — CHANGES_REQUESTED** [review]

File: src/locales/vi/location/state.ts
Comment (ST-DDT) [reply src/locales/vi/location/state.ts:1]: If the real names have them, then yes.
If the user doesnt want them they can still strip them.

Comment (ST-DDT) [reply src/locales/vi/location/state.ts:1]: @asharnadeem Can you please address the feedback?

Comment (ST-DDT) [reply src/locales/vi/location/state.ts:1]: @asharnadeem Are you still interested in this PR?
If yes, please address the feedback. If not, please close it.

Comment (matthewmayer) [reply src/locales/vi/location/state.ts:1]: i went ahead and added the diacritics in f9b51fbe

**Review by matthewmayer — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**